src/outputs/parthenon_xdmf.cpp

```diff
@@ -106,52 +114,80 @@ void genXDMF(std::string hdfFile, Mesh *pm, SimTime *tm, IndexDomain domain, int
     xdmf << R"(    <Time Value=")" << tm->time << R"("/>)" << std::endl;
   }
 
-  std::string blockTopology =
-      R"(      <Topology TopologyType="3DRectMesh" Dimensions=")" +
-      std::to_string(nx3 + 1) + " " + std::to_string(nx2 + 1) + " " +
-      std::to_string(nx1 + 1) + R"("/>)" + '\n';
-  const std::string slabPreDim = R"(        <DataItem ItemType="HyperSlab" Dimensions=")";
-  const std::string slabPreBlock2D =
-      R"("><DataItem Dimensions="3 2" NumberType="Int" Format="XML">)";
-  const std::string slabTrailer = "</DataItem>";
-
   // Now write Grid for each block
+  int ndim;
   dims[0] = pm->nbtotal;
+  const int n3_offset = output_coords ? (nx3 > 1) : 1;
+  const int n2_offset = output_coords ? (nx2 > 1) : 1;
+  int ndim_mesh;
+  std::string mesh_type, dimstring;
+  if (output_coords) {
+    if (nx3 > 1) {
+      ndim_mesh = 3;
+      mesh_type = "3DSMesh";
+      dimstring = StringPrintf("%d %d %d", nx3 + n3_offset, nx2 + n2_offset, nx1 + 1);
+    } else if (nx2 > 1) {
+      ndim_mesh == 2;
+      mesh_type = "2DSMesh";
+      dimstring = StringPrintf("%d %d", nx2 + n2_offset, nx1 + 1);
+    } else {
+      ndim_mesh == 1;
+      mesh_type = "Polyline";
+      dimstring = StringPrintf("%d", nx1 + 1);
+    }
+  } else {
+    mesh_type = "3DRectMesh";
+    ndim_mesh = 3;
+    dimstring = StringPrintf("%d %d %d", nx3 + n3_offset, nx2 + n2_offset, nx1 + 1);
+  }
```

Comment:
This madness merits explanation. The story goes like this:
- We want Visit/Paraview to use output mesh data (rather than the `x*f` arrays) as coordinates. This means going to `SMesh` in 3D and 2D.
- Paraview/Visit expects coordinate positions to be at nodes. We always output 1 extra face even in trivial directions for the `x*f` arrays. But we don't for positions of node-centered fields. This necessitates changing the dimension of the data that is read by visit/paraview depedning on output dimension. In other words, in 2D we must use `2DSMesh`, not `3DSMesh` and it needs an `(nx2+1, nx1+1)` array, not a `(1, nx2+1, nx1+1)` array.
- 1D gets crazier as there is no structured mesh object in 1D in `xdmf`. Only `Polyline`.

The remaining changes to this file can be understood as contorting themselves to support the above.

Reply:
Ok I am dropping the 1D as it's a lot of contortion for an uncommon use-case.